boards/stm32h743zi/src/tasks/mqtt_send_and_recv.rs

```diff
@@ -0,0 +1,140 @@
+use crate::{log::log, config::GATEWAY_ADDRESS};
+
+use embassy_net::{tcp::TcpSocket, Ipv4Address, Stack};
+use embassy_stm32::{
+    eth::{generic_smi::GenericSMI, Ethernet},
+    peripherals::ETH,
+};
+use embassy_sync::{blocking_mutex::raw::ThreadModeRawMutex, channel::Channel};
+use embassy_time::{Duration, Timer};
+use hyped_core::{
+    format,
+    format_string::show,
+    log_types::LogLevel,
+    mqtt::{HypedMqttClient, MqttMessage},
+};
+
+use {defmt_rtt as _, panic_probe as _};
+
+/// Channel for sending messages to the MQTT broker
+pub static SEND_CHANNEL: Channel<ThreadModeRawMutex, MqttMessage, 128> = Channel::new();
+
+/// Task for receiving messages from the MQTT broker
+#[embassy_executor::task]
+pub async fn mqtt_recv_task(stack: &'static Stack<Ethernet<'static, ETH, GenericSMI>>) {
+    let mut rx_buffer: [u8; 4096] = [0; 4096];
+    let mut tx_buffer: [u8; 4096] = [0; 4096];
+    let mut socket = TcpSocket::new(&stack, &mut rx_buffer, &mut tx_buffer);
+    socket.set_timeout(Some(embassy_time::Duration::from_secs(600)));
+    log(LogLevel::Info, "Connecting to Receive Socket...").await;
+    match socket
+        .connect(GATEWAY_ADDRESS)
```

Comment:
This should be called `MQTT_BROKER_ADDRESS`. I think the was Kshitij had networking configured meant that the same IP was used for both